ppg-21#ValiderEtape

ibib-git opened the pull request:
ppg21 & ppg19

ibib-git committed:
PPG-21 && PPG-19
valider-tache.service.ts : creation d'un service pour valider une tache
utilisateur.service.ts : suppression attribut non utilisé
TacheModel.ts : correction orthographe
utilisateur-interface.component.ts : ajout initialiisation utilisateur
tache-etape-workflow.component : ajout d'un template pour la modal de detail d'une tache
etape-workflow.component : ajout de trois input utile au tache-etape-workflow.component (membresProjet, idEtapeSuivante, idDerniereEtape)
workflow.component : ajout des méthodes getIdDerniereEtape() et getIdEtapeSuivante(idEtapeCourante) pour le passage en input au etape-workflow.component
app.module.ts : ajout du pipe ordreEtapePipe

ibib-git committed:
PPG-21 && PPG-19
tache-etape-workflow.component & etape-workflow.component & workflow.component : ajout des output pour l'update du projet

WattecampsProg commented on src/app/app-routing.module.ts:
```diff
@@ -17,7 +17,7 @@ const routes: Routes = [
   {path: 'dashboard/Workflow/:id', component: WorkflowComponent},
   {path: 'projet/gestion', component: ProjetGestionComponent},
-  {path: 'utilisateur/:id',component: UtilisateurInterfaceComponent},
+  {path: 'utilisateur/:id', component: UtilisateurInterfaceComponent},
```
.......... espace .............

WattecampsProg commented on src/app/features/workflow/etape-workflow/etape-workflow.component.html:
```diff
@@ -13,12 +13,12 @@ <h5>{{etape.nom}}</h5>
     <nb-card-footer >
         <div class="etapeFooter">
-            <button nbButton>Créer une tache</button>
+            <button disabled nbButton>Créer une tache</button>
```
Je modifierai ça

WattecampsProg commented on src/app/features/workflow/etape-workflow/etape-workflow.component.ts:
```diff
@@ -58,4 +68,8 @@ export class EtapeWorkflowComponent implements OnInit {
     );
   }
 
```
J'ai fais pariel

ibib-git replied:
pour le output tu parle?

WattecampsProg commented on src/app/features/workflow/workflow.component.ts:
```diff
@@ -31,8 +39,26 @@ export class WorkflowComponent implements OnInit {
         },
         () => {
           this.toastrServ.danger('Erreur dans le chargement du projet ', 'Workflow', {[status]: 'danger'});
+          this.routerServ.navigateByUrl('');
```
Il va vers home ou un Oubli ?

ibib-git replied:
il redirige vers la page par defaut, dans notre cas c'est home

ibib-git committed:
PPG-21 && PPG-19
merge fiw bugs : ParticipationModel.ts & ParticipationDetailModel.ts : ajout d'un attribut isActif par Thomas en back
utilisateur.service.ts : correction adresse

ibib-git committed:
PPG-21 && PPG-19
merge fiw bugs :
ordre-etape.pipe.ts : modification du type de retour de unknow en EtapeWorkflowModel[]
workflow.component.ts : etapeTrieeOrdre en variable global pour réutilisation car besoin de ne trier que 1 fois, modification des initialisations (etapeTrieeOrdre et le projet seront rempli dans l appel au service getProjet), getIdEtapeSuivante : modification de la ternaire car besoin de juste tester la limite du [] dans le cas ou je suis en dernière colonne

ibib-git committed:
PPG-21 && PPG-19
merge fiw bugs : modifierprojet.service.ts changer type de retour
projet-gestion.component.ts : initialisation du projet et rechargement de la page
